lib/index.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+
+const request = require('request-promise-native');
+const cache = require('./cache');
+
+const defaultInt = 1000;
+
+module.exports = endpoints => {
+
+  return (req, res, next) => {
+    if (endpoints && endpoints.length) {
+
+      let requests = endpoints.reduce((memo, option) => {
+        const url = option.url;
+        const interval = parseInt(option.interval, 10) || defaultInt;
+        const currTime = Date.now();
+        const futureTime = currTime + interval;
+        const savedTime = cache.get(url);
+
+        // Is this the first time or has the interval expired
+        if (isNaN(savedTime) || savedTime < currTime) {
+          cache.set(url, futureTime);
+          memo.push(request(url).promise());
+        }
+        return memo;
+
+      }, []);
+
+      // Process all pending requests
+      return Promise.all(requests)
+        .then(resolved => {
```

Comment:
I don't think we can rely on just a response here. As lenny states we should rely on the status code when we get a response. It's the norm.